### ISSUE TITLE
variable of type string can be filled with an array of a single string without being validated as faulty

### ISSUE THREAD

**Issue by Akivashi** [typestack/class-validator#394]
If I have the following class:
```import { IsNotEmpty } from "class-validator";

export default class ExampleClass {
  @IsNotEmpty()
  private exampleValue: string = "";

  public getExampleValue = (): string => this.exampleValue;
}
```
And i try to perform the following code:
```
      const plainObject = {
        exampleValue: ["this is a string in an array"]
      }
      var classObject: ExampleClass = plainToClass(ExampleClass , plainObject);
      const errors = await validate(classObject, {validationError: { target: false, value: false }});
      console.log(errors);
      console.log(classObject.getExampleValue ())
```
No errors will be found during the validation of the plain object.
The code however still sees the `exampleValue` to be of type `string`.
If this goes unnoticed, this can lead to issues with for example string manipulation (e.g. methods such as `substr` or `slice`) in code that should only be run after validation of the plain object.
In short: a validation error should be provided for this particular issue.

If i can find some time, i might look into this myself soon.

**Comment by vlapo** [comment]
From docs: 
> @IsNotEmpty() | Checks if given value is not empty (!== '', !== null, !== undefined).

So from this point of view do not get any validation error is correct behaviour. Use `@IsString()` decorator too.

```ts
export default class ExampleClass {
  @IsString()
  @IsNotEmpty()
  private exampleValue: string = "";

  // ... 
}
```

**Comment by Akivashi** [comment]
I was under the impression that primitive types would be validated automatically. I apparently skipped over that notion when reading the documentation. My bad, i'll close the issue.
